src/views/Player.vue

```diff
@@ -346,6 +350,27 @@ export default defineComponent({
       resetConfetti();
     }
 
+    async function fetchQuestionBucket(requestedQuestionIndex: number) {
+      const bucketToFetch = Math.floor(requestedQuestionIndex / store.state.bucketSize)
+      const bucketStartIndex = store.state.questionsBucketingMap[bucketToFetch].bucketStartIndex
+      const bucketEndIndex = store.state.questionsBucketingMap[bucketToFetch].bucketEndIndex
+
+      const fetchedQuestions = await QuestionAPIService.getQuestionsByQuestionSet(
+        state.questions[requestedQuestionIndex].question_set_id,
+        bucketStartIndex,
+        store.state.bucketSize
+      )
+
+      for (let i = bucketStartIndex; i <= bucketEndIndex; i++) {
+        state.questions[i] = fetchedQuestions[i - bucketStartIndex]
+      }
```

Comment:
This will be wrong
Let's say I want to get the 2nd bucket.
First bucket was questions 0th index - 9th index (10 questions) which come by default when doing a GET on quiz.

Second bucket will be from 10th index - 19th index. This index is of the main `questions` array.
So whatever we fetch from the db, needs to be filled into 10th, 11th, 12th ... 19th indices in the questions array.

If I wrote what you have suggested, `state.questions[i]` - here `i` will start from 0 so it will replace the questions of the first subset in the main `questions` array